src/runtime/scheduler/group.rs

```diff
@@ -51,170 +49,140 @@ pub struct TaskGroup {
 
-        // Initialize the appropriate page offset.
-        let (waker_page_ref, waker_page_offset): (&WakerPageRef, usize) = {
-            let (waker_page_index, waker_page_offset) = self.get_waker_page_index_and_offset(pin_slab_index)?;
-            (&self.waker_page_refs[waker_page_index], waker_page_offset)
-        };
-        waker_page_ref.initialize(waker_page_offset);
+        let (i, offset) = self.locate(idx)?;
+        self.pages[i].initialize(offset);
 
-        trace!("insert(): name={:?}, pin_slab_index={:?}", task_name, pin_slab_index);
+        trace!("insert(): name={:?}, idx={:?}", name, idx);
 
-        // Poll new task.
-        match self.poll_runnable_task(pin_slab_index) {
+        match self.run_task(idx) {
             Some(task) => Some(InsertResult::Completed(task)),
-            None => Some(InsertResult::Inserted(pin_slab_index.into())),
+            None => Some(InsertResult::Inserted(idx.into())),
         }
     }
 
-    pub fn get_mut_task(&mut self, pin_slab_index: usize) -> Option<Pin<&mut Box<dyn Task>>> {
-        self.tasks.get_pin_mut(pin_slab_index)
+    pub fn get_task_mut(&mut self, idx: usize) -> Option<Pin<&mut Box<dyn Task>>> {
+        self.tasks.get_pin_mut(idx)
     }
 
     /// Computes the page and page offset of a given task based on its total offset.
-    fn get_waker_page_index_and_offset(&self, pin_slab_index: usize) -> Option<(usize, usize)> {
+    fn locate(&self, pin_slab_index: usize) -> Option<(usize, usize)> {
         // This check ensures that slab slot is actually occupied but trusts that pin_slab_index is for this task.
         if !self.tasks.contains(pin_slab_index) {
             return None;
         }
-        let waker_page_index: usize = pin_slab_index >> WAKER_BIT_LENGTH_SHIFT;
-        let waker_page_offset: usize = Self::get_waker_page_offset(pin_slab_index);
-        Some((waker_page_index, waker_page_offset))
+        let index = pin_slab_index >> WAKER_BIT_LENGTH_SHIFT;
+        let offset = Self::get_page_offset(pin_slab_index);
+        Some((index, offset))
     }
 
-    /// Add new page(s) to hold this future's status if the current page is filled. This may result in addition of
-    /// multiple pages because of the gap between the pin slab index and the current page index.
-    fn add_new_pages_up_to_pin_slab_index(&mut self, pin_slab_index: usize) {
-        while pin_slab_index >= (self.waker_page_refs.len() << WAKER_BIT_LENGTH_SHIFT) {
-            self.waker_page_refs.push(WakerPageRef::default());
+    /// Grow pages to fit the given pin slab index.
+    fn grow_pages(&mut self, idx: usize) {
+        while idx >= (self.pages.len() << WAKER_BIT_LENGTH_SHIFT) {
+            self.pages.push(WakerPageRef::default());
         }
     }
 
-    pub fn get_num_waker_pages(&self) -> usize {
-        self.waker_page_refs.len()
-    }
-
-    fn get_waker_page_offset(pin_slab_index: usize) -> usize {
+    fn get_page_offset(pin_slab_index: usize) -> usize {
```

Comment:
should we get rid of the get_ on these function names as well and just name it page_offset?